dotcom-rendering/src/web/components/Lightbox.tsx

```diff
@@ -0,0 +1,591 @@
+import { css, Global } from '@emotion/react';
```

Comment:
I don't feel 100% sure that server-side-rendering this is the best approach, we're adding a lot of HTML & CSS to the DOM for everyone that is used only a for a feature that is used by few. Is there a way we could client side render this instead? I wonder if we were to client render this whether we'd be getting as much value from keeping the JS as vanilla instead of react - but I'm not familiar enough with the rest of the code to be confident

Reply:
Yeah, you're not wrong. Ideally we'd not serve unused html and be able to inject it as needed 

I used this approach because it's what Frontend does and also because _knowing_ the lightbox is in the dom simplifies things. It's also a trade off. Lazy loading the html and js would create a waterfall of requests on click whereas opening the lightbox now is pretty fast as the html is ready to go.

Re the ability to make things more reacty if it was all client side, I actually find it easier to use vanilla javascript when dealing with dom updates and event listeners. This is maybe a personal thing but I find the React lifecycle can get in the way in these sorts of situations? So I'd personally still try to use the same approach even if everything were client-side. It's possible that wouldn't work out though 🤷

Reply:
> I don't feel 100% sure that server-side-rendering this is the best approach, we're adding a lot of HTML & CSS to the DOM for everyone that is used only a for a feature that is used by few. Is there a way we could client side render this instead? I wonder if we were to client render this whether we'd be getting as much value from keeping the JS as vanilla instead of react - but I'm not familiar enough with the rest of the code to be confident

This comment was really insightful and highlighted a very real problem. The size of the html generated on the server was actually very large. The `Picture` element returns a lot and once you've got multiple images the size of the lightbox html string can reach 90Kb 😱 

Unfortunately, it triggered a React vs. Javascript brouhaha that wasn't especially helpful or relevant to the problem 😺 

Anyway, returning to the _problem_, I've removed the array of `images` out of the lightbox html that gets generated on the server and broken it out into `LIghtboxImages`. This component is then dynamically inserted into the dom during hydration using `React.createPortal`.

This change resulted into two additional improvements:

1) I added `LightboxLoader`. This will improve the UX for people on slow connections/devices now that we have client side rendering

2) I changed the type for the data inserted into the model from a superset of `ImageBlockElement` to a _much smaller_ subset of this data. This was required because the array of `ImageBlockElement` images was also in the region of 90Kb 😬 and I needed to serialise this array into the dom to give to `LightboxImages`.

So we now have 

```
// The server rendered html for lightbox
// 2Kb
<div id="gu-lightbox" aria-modal="true" role="dialog"> ... <ul placeholder> {**empty**} </ul> ... </div>
// The island which hydrates lightbox
// ~1-3Kb
<gu-island name="LightboxJavascript" deferuntil="hash" props="{ a much smaller data object }" clientonly="true"></gu-island>
```